File: LeetCode/Kai/1TwoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] result = new int[2];
        
        for(int i=0; i<nums.length;i++) {
            if(map.containsKey(nums[i])) {
                result[0] = map.get(nums[i])+1;
                result[1] = i+1;
                break;
            } else {
                map.put(target-nums[i], i);
            }
        }
        return result;
    }
    
}

File: LeetCode/Lei/1TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        map<int, int> m;
        for(int i=0; i<int(nums.size()); i++)
        {
          if(m.find(nums[i])==m.end())
          {
            m[target-nums[i]]=i;
          }
          else
          {
            result.push_back(m[nums[i]]+1);
            result.push_back(i+1);
            break;
          }
        }
    return result;
    }
};
